.github/workflows/validate-generated-files.yml

```diff
@@ -0,0 +1,24 @@
+name: validate-generated-files
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  validate-generated-files:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Check generated files
+        run: |
+          export PATH=$PATH:$(go env GOPATH)/bin
+          make install-tools generate
```

Comment:
Btw, we could also update the `install-tools` target (to [this](https://github.com/ConduitIO/conduit-connector-template/blob/1c920dda7668fc1a53fd6c991d78708dc658a032/Makefile#L26)), the updated target uses exactly the dependencies that the tool needs, otherwise it takes whatever is in `go.mod` which could fail the build.